### ISSUE TITLE
Report EPIPE write EPIPE

### ISSUE THREAD

**Issue by kwangle912** [vknabel/vscode-swiftlint#11]
`Error: write EPIPE
    at afterWriteDispatched (internal/stream_base_commons.js:146:25)
    at writeGeneric (internal/stream_base_commons.js:137:3)
    at Socket._writeGeneric (net.js:698:11)
    at Socket._write (net.js:710:8)
    at doWrite (_stream_writable.js:418:12)
    at writeOrBuffer (_stream_writable.js:402:5)
    at Socket.Writable.write (_stream_writable.js:302:11)
    at Socket.Writable.end (_stream_writable.js:583:10)
    at Socket.end (net.js:536:31)
    at c:\Users\Dell\.vscode\extensions\vknabel.vscode-swiftlint-1.1.0\out\lint.js:174:24
    at new Promise (<anonymous>)
    at execSwiftlint (c:\Users\Dell\.vscode\extensions\vknabel.vscode-swiftlint-1.1.0\out\lint.js:148:12)
    at Object.<anonymous> (c:\Users\Dell\.vscode\extensions\vknabel.vscode-swiftlint-1.1.0\out\lint.js:28:38)
    at Generator.next (<anonymous>)
    at fulfilled (c:\Users\Dell\.vscode\extensions\vknabel.vscode-swiftlint-1.1.0\out\lint.js:5:58)`

**Comment by vknabel** [comment]
@kwangle912 thanks for the report! Do you still remember why/when this error happened?

**Comment by vknabel** [comment]
@yangjunying did also experienced this issue in #15

Do you still remember when this error happened?
Was there anything special about your local setup?
How big was your project?
Does it include dependencies like CocoaPods or Carthage?

**Comment by zachhardesty7** [comment]
@vknabel Four issues with no followups :/
I believe I saw the `EPIPE` error the first time I tried to open a Swift file after downloading this extension. Unfortunately, I can't get it to happen again but had a reproducible different error right afterwards. This same result occurred from running `swiftlint` via CLI.

```
Error: Command failed: /usr/local/bin/swiftlint --quiet --reporter json --use-stdin
Fatal error: Loading sourcekitd.framework/Versions/A/sourcekitd failed: file /private/tmp/swiftlint-20200404-90819-1vnlirh/.build/checkouts/SourceKitten/Source/SourceKittenFramework/library_wrapper.swift, line 39
	at ChildProcess.exithandler (child_process.js:308:12)
    at ChildProcess.emit (events.js:203:13)
    at maybeClose (internal/child_process.js:1021:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
```

I don't know if this is directly related to their errors, but it might help you find the cause of the issue or anybody else who gets this error when trying to run this extension. I could reproduce it by running: `sudo xcode-select -s /Library/Developer/CommandLineTools`.

For me, the above error was resolved by this suggestion:

> @jjkaufman, i found the solution in another question, i ran this with the path of my xcode
> `sudo xcode-select -s /Applications/Xcode.app/Contents/Developer`
> 
> _Originally posted by @ralcr in https://github.com/realm/SwiftLint/issues/2937#issuecomment-554273231_

**Comment by vknabel** [comment]
@zachhardesty7 wow. No chance I ever figured out!
Thank you very much! Now I will be able to work on a bugfix.

**Comment by vknabel** [comment]
Fix has been published as [1.2.4](https://github.com/vknabel/vscode-swiftlint/releases/tag/1.2.4)
